packages/fluentui/react-northstar/src/components/Carousel/CarouselNavigationItem.tsx

```diff
@@ -57,92 +69,142 @@ export interface CarouselNavigationItemProps extends UIComponentProps, ChildrenC
   thumbnails?: boolean;
 }
 
+export type CarouselNavigationItemStylesProps = Required<
+  Pick<CarouselNavigationItemProps, 'thumbnails' | 'vertical' | 'active' | 'iconOnly' | 'primary'>
+> & {
+  hasContent: boolean;
+  hasIndicator: boolean;
+};
+
 export const carouselNavigationItemClassName = 'ui-carousel__navigationitem';
 export const carouselNavigationItemSlotClassNames: CarouselNavigationItemSlotClassNames = {
   indicator: `${carouselNavigationItemClassName}__indicator`,
   content: `${carouselNavigationItemClassName}__content`,
 };
 
-class CarouselNavigationItem extends UIComponent<WithAsProp<CarouselNavigationItemProps>> {
-  static displayName = 'CarouselNavigationItem';
-
-  static deprecated_className = carouselNavigationItemClassName;
-
-  static create: ShorthandFactory<CarouselNavigationItemProps>;
-
-  static propTypes = {
-    ...commonPropTypes.createCommon(),
-    active: PropTypes.bool,
-    indicator: customPropTypes.shorthandAllowingChildren,
-    iconOnly: PropTypes.bool,
-    index: PropTypes.number,
-    onClick: PropTypes.func,
-    primary: customPropTypes.every([customPropTypes.disallow(['secondary']), PropTypes.bool]),
-    secondary: customPropTypes.every([customPropTypes.disallow(['primary']), PropTypes.bool]),
-    vertical: PropTypes.bool,
-    thumbnails: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    accessibility: tabBehavior as Accessibility,
-    as: 'li',
-    indicator: {},
-  };
-
-  renderContent({ content, indicator, styles }) {
+export const CarouselNavigationItem: React.FC<WithAsProp<CarouselNavigationItemProps>> &
+  FluentComponentStaticProps<CarouselNavigationItemProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(CarouselNavigationItem.displayName, context.telemetry);
+  setStart();
+
+  const {
+    children,
+    thumbnails,
+    vertical,
+    active,
+    content,
+    iconOnly,
+    primary,
+    indicator,
+    className,
+    design,
+    styles,
+    variables,
+  } = props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(CarouselNavigationItem.handledProps, props);
+
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: CarouselNavigationItem.displayName,
+    actionHandlers: {
+      performClick: event => !event.defaultPrevented && handleClick(event),
+    },
+  });
+
+  const { classes, styles: resolvedStyles } = useStyles<CarouselNavigationItemStylesProps>(
+    CarouselNavigationItem.displayName,
+    {
+      className: carouselNavigationItemClassName,
+      mapPropsToStyles: () => ({
+        thumbnails,
+        vertical,
+        active,
+        hasContent: !!content && !_.isEmpty(content),
+        iconOnly,
+        primary,
+        hasIndicator: !!indicator && !_.isEmpty(indicator),
```

Comment:
```suggestion
        hasIndicator: !!indicator,
```

Reply:
`{}` as an indicator means render default, false and null are used for do not render anything, so just this condition is enough